packages/dataviews/src/dataviews-layouts/table/use-is-scrolled-end.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+import { isRTL } from '@wordpress/i18n';
+
+const isScrolledToEnd = ( element: Element ) => {
+	if ( isRTL() ) {
+		const scrollLeft = Math.abs( element.scrollLeft );
+		return scrollLeft <= 1;
+	}
+
+	return element.scrollLeft + element.clientWidth >= element.scrollWidth - 1;
+};
+
+export function useIsScrolledEnd( {
+	selector,
+	enabled = false,
+}: {
+	selector: string;
+	enabled?: boolean;
+} ): boolean {
+	const [ isScrolledEnd, setIsScrolledEnd ] = useState( false );
+
+	useEffect( () => {
+		if ( typeof window === 'undefined' || ! enabled ) {
+			return () => {};
+		}
+
+		const scrollContainer = document.querySelector( selector );
```

Comment:
It's possible to hold a stale reference, or this may be `null` depending on when the DOM element is committed. It would be best to pass a ref.

Reply:
Yeah, it’s not a perfect way for getting the scroll container, but accessing the ref directly isn’t straightforward. Here are a couple of alternatives I considered:

* Expose the `containerRef` via context so it’s easily accessible wherever it’s needed. However, the `containerRef` might only be needed in this specific place, so adding it to context could be unnecessary overhead.
* Pass the table’s ref and use [getScrollContainer](https://github.com/WordPress/gutenberg/blob/72c922b679cbc0d83220514165ea66837de131ca/packages/dom/src/dom/get-scroll-container.js#L15) to locate the scroll container. That said, I’m not sure it’s worth introducing the @wordpress/dom package just for this.

Any thoughts?

Reply:
> Expose the containerRef via context so it’s easily accessible wherever it’s needed. However, the containerRef might only be needed in this specific place, so adding it to context could be unnecessary overhead.

It should be very lightweight, seeing that it'll only save a reference to an object already in memory. I don't have any reservations about adding that reference to the context. 

The trade-off seems worth it to me. Passing the selector as a string seems like a less desirable outcome.

Reply:
OK. I've added the `containerRef` to the context